upcloud/client/client.go

```diff
@@ -139,6 +169,50 @@ func (c *Client) getBaseURL() string {
 	return fmt.Sprintf("%s/%s", c.config.baseURL, APIVersion)
 }
 
+// Pretty prints given JSON bytes. If the JSON is not valid, returns the original bytes as string.
+func prettyJSON(i []byte) string {
+	var o bytes.Buffer
+	if err := json.Indent(&o, i, "", "  "); err != nil {
+		return string(i)
+	}
+	return o.String()
+}
+
+func (c *Client) logRequest(r *http.Request, body []byte) {
+	const authorization string = "Authorization"
+
+	if c.config.logger != nil {
+		headers := r.Header.Clone()
+		if _, ok := headers[authorization]; ok {
+			auth := strings.Split(headers.Get(authorization), " ")
+			// Redact the token part of the Authorization header or the whole value if value has no space found.
+			if len(auth) > 1 {
+				headers.Set(authorization, fmt.Sprintf("%s xxxxx", auth[0]))
```

Comment:
Some form of "redacted" could work here as well.

```suggestion
				headers.Set(authorization, fmt.Sprintf("%s [REDACTED]", auth[0]))
```

Reply:
I chose `xxxxx` to be in sync with [URL.Redacted](https://pkg.go.dev/net/url#URL.Redacted), but `[REDACTED]` would maybe cause less confusion 🤔

Reply:
Let's go with `[REDACTED]` as it provider more information